packages/grafana-ui/src/components/OptionsUI/number.tsx

```diff
@@ -36,7 +36,7 @@ export const NumberValueEditor: React.FC<FieldConfigEditorProps<number, NumberFi
         );
       }
     },
-    [onChange]
+    [onChange, settings?.integer]
```

Comment:
nit: Best practices is to destruct nested props and use them in the dependencies.

Reply:
so I was not sure about this because settings can be undefined, how would you rewrite this?